index.js

```diff
@@ -1580,6 +1581,78 @@ class GithubScm extends Scm {
             throw err;
         });
     }
+
+    /**
+    * Open a pull request on the repository with given file change
+    *
+    * @method _openPr
+    * @param  {Object}     config                  Configuration
+    * @param  {String}     config.checkoutUrl      Checkout url to the repo
+    * @param  {String}     config.token            Service token to authenticate with the SCM service
+    * @param  {String}     config.files            Files to open pull request with
+    * @param  {String}     config.title            Pull request title
+    * @param  {String}     config.message          Pull request message
+    * @param  {String}     [config.scmContext]     The scm context name
+    * @return {Promise}                            Resolves when operation completed without failure
+    */
+    async _openPr(config) {
+        const { checkoutUrl, token, files, title, message } = config;
+        const [, , owner, repo, branch] = checkoutUrl.match(CHECKOUT_URL_REGEX);
+        const newBranch = title.replace(/ /g, '_');
+
+        return this.breaker.runCommand({
```

Comment:
might be better to do try catch

Reply:
does it really matter if all we going to do is log error here? one promise catch at the end of chain will do the work instead of multiple try catch block

Reply:
okay